pkg/apis/core/validation/validation.go

```diff
@@ -3682,6 +3701,57 @@ func validateHostUsers(spec *core.PodSpec, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+// validateFileKeyRefVolume validates that volumes referenced by FileKeyRef environment variables
+// are of type emptyDir. FileKeyRef requires emptyDir volumes to ensure proper file access.
+func validateFileKeyRefVolume(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Collect all volume names referenced by FileKeyRef environment variables
+	referencedVolumeNames := sets.Set[string]{}
```

Comment:
Rather than a set which usually holds 0 or 1 items, how about using a slice for this and starting it as nil.  You can use `slices.Contains` to check it